pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -1250,3 +1295,241 @@ func (r *Builder) GetPopulatorTaskName(pvc *core.PersistentVolumeClaim) (taskNam
+				err = r.Destination.Client.Delete(context.Background(), scratchPVC, &client.DeleteOptions{})
+				if err != nil {
+					r.Log.Error(err, "Failed to delete scratch PVC", "pvc", scratchPVC.Name)
+				}
+			case batchv1.JobFailed:
+				if convertJob.Status.Failed >= 3 {
+					return true, liberr.New("convert job failed")
+				}
+			}
+		}
+	}
+
+	if completedPVCs == len(pvcs) {
+		r.Log.Info("All PVCs completed")
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (r *Builder) ensureJob(pvc *core.PersistentVolumeClaim, jobType, srcFormat, dstFormat string) (*batchv1.Job, error) {
+	jobName := getJobName(pvc, jobType)
+	job := &batchv1.Job{}
+	err := r.Client.Get(context.Background(), client.ObjectKey{Name: jobName, Namespace: pvc.Namespace}, job)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			if jobType == "convert" {
+				job := createConvertJob(pvc, srcFormat, dstFormat, r.getLabels(pvc.Annotations["vmID"]))
+				err = r.Destination.Client.Create(context.Background(), job, &client.CreateOptions{})
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+		return nil, err
+	}
+
+	return job, err
+}
+
+func createConvertJob(pvc *core.PersistentVolumeClaim, srcFormat, dstFormat string, labels map[string]string) *batchv1.Job {
+	return &batchv1.Job{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      fmt.Sprintf("convert-%s", pvc.Name),
+			Namespace: pvc.Namespace,
+			Labels:    labels,
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit: pointer.Int32(3),
+			Completions:  pointer.Int32(1),
+			Template: core.PodTemplateSpec{
+				Spec: core.PodSpec{
+					SecurityContext: &core.PodSecurityContext{
+						SeccompProfile: &core.SeccompProfile{
+							Type: core.SeccompProfileTypeRuntimeDefault,
+						},
+						FSGroup: pointer.Int64(107),
+					},
+					RestartPolicy: core.RestartPolicyNever,
+					Containers: []core.Container{
+						makeConversionContainer(pvc, srcFormat, dstFormat),
+					},
+					Volumes: []core.Volume{
+						{
+							Name: "source",
+							VolumeSource: core.VolumeSource{
+								PersistentVolumeClaim: &core.PersistentVolumeClaimVolumeSource{
+									ClaimName: pvc.Name,
+								},
+							},
+						},
+						{
+							Name: "target",
+							VolumeSource: core.VolumeSource{
+								PersistentVolumeClaim: &core.PersistentVolumeClaimVolumeSource{
+									ClaimName: getScratchPVCName(pvc),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func makeConversionContainer(pvc *core.PersistentVolumeClaim, srcFormat, dstFormat string) core.Container {
+	container := core.Container{
+		Name:  "convert",
+		Image: base.Settings.VirtV2vImageCold,
+		SecurityContext: &core.SecurityContext{
+			AllowPrivilegeEscalation: pointer.Bool(false),
+			RunAsNonRoot:             pointer.Bool(true),
+			RunAsUser:                pointer.Int64(107),
+			Capabilities: &core.Capabilities{
+				Drop: []core.Capability{"ALL"},
+			},
+		},
+		Command: []string{"/usr/local/bin/image-converter"},
```

Comment:
do we really need image-converter? what I had in mind is that by using the existing image, we can extend its `entrypoint` script to initiate conversion using qemu-img rather than initiating virt-v2v...

Reply:
I know, I just didn't want to mess with the bash script and risk breaking ova and vsphere... also it would've been easier if the virt-v2v part would've been wrapped too, and not broken into go (for progress monitoring) and bash

Reply:
yeah, [we reused the original script we had before moving the virt-v2v image to this repository](https://github.com/kubev2v/forklift-virt-v2v/blob/main/bin/entrypoint) - but we could have triggered virt-v2v with go.
ok, I see no good reason to prefer a script over this code so let's keep it